src/index.js

```diff
@@ -79,9 +83,42 @@ export default class MarkersPlugin extends UICorePlugin {
       internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
     }
     internalMarker.onDestroy()
+    this._markers.splice(index, 1)
```

Comment:
.indexOf(marker) would never work since its very rare to have the exact same marker instance lying around to invoke removeMarker method. I didn't want to change removeMarker to use index since it would break some poor user out there. _removeInternalMarker is an internal method anyway and hence we can freely pass whatever variables we want instead of forcing it to do lookups?